src/game/engine/internals/CollisionDetector.js

```diff
@@ -31,19 +29,17 @@ class CollisionDetector {
   }
 
   detectCollisions() {
-    /** @type {import('../../shared').Collision[]} */
-    const collisions = [];
     this.#colliders.forEach((collider1) => {
       this.#colliders.forEach((collider2) => {
-        if (collider1 !== collider2 && collider1.hasCollisionWith(collider2)) {
-          collisions.push(new Collision(collider1.gameObject, collider2.gameObject));
+        if (collider1 !== collider2) {
+          const collision = collider1.hasCollisionWith(collider2);
+
+          if (collision) {
+            collision.gameObject1.onCollision(collision, collision.gameObject2);
```

Comment:
Nie wiem czy ta linijka powinna być w tym miejscu :man_shrugging: ? Problem jest taki, że `onCollision` może zmienić położenie obiektu, przez co jeszcze nieobliczone kolizje mogą "zniknąć", chociaż się wydarzyły. Dlatego wcześniej obliczałem `onCollision` po wyznaczeniu wszystkich kolizji (po `forEach`).

Reply:
Uznałem że to to nie jest szczególnie duży problem, bo teraz przechodząc przez te bloki które wcześniej "znikały" i tak `resolutionVector` będzie równy `Vector.Zero` (bo jeśli jakiś był pominięty, to znaczy że poprzedni `onCollision` wyniósł nas poza ten blok), więc logika kolizji nic nie zrobi.

Ale to rozwiązanie jest faktycznie bardziej future proof, bo pojawią nam się kolizje które nie zmieniają przemieszczenia (np. drzwi) 👍